compiler/parser/src/fstring.rs

```diff
@@ -105,19 +106,16 @@ impl<'a> FStringParser<'a> {
                                 nested -= 1;
                                 if nested == 0 {
                                     formatted_value_piece.push(next);
-                                    spec_constructor.push(
-                                        self.expr(ExprKind::FormattedValue {
-                                            value: Box::new(
-                                                FStringParser::new(
-                                                    &formatted_value_piece,
-                                                    Location::default(),
-                                                    &self.recurse_lvl + 1,
-                                                )
-                                                .parse()?,
-                                            ),
-                                            conversion: ConversionFlag::None as _,
-                                            format_spec: None,
-                                        }),
+                                    let values = FStringParser::new(
+                                        &formatted_value_piece,
+                                        Location::default(),
+                                        &self.recurse_lvl + 1,
+                                    )
+                                    .parse()?;
+                                    spec_constructor.push(values
+                                        .into_iter()
+                                        .exactly_one()
```

Comment:
I worried if this is possible to be broken, but `ExpressionNestedTooDeeply` seems to prevent it successfully.